lib/fdotypes.c

```diff
@@ -5490,20 +5490,214 @@ bool fdo_compare_rv_lists(fdo_rendezvous_list_t *rv_list1,
 			  fdo_rendezvous_list_t *rv_list2)
 {
 	bool retval = false;
+	int rv_directive_index = 0;
+	int rv_instr_index = 0;
+	fdo_rendezvous_list_t *rv_list1_traverse = rv_list1;
+	fdo_rendezvous_list_t *rv_list2_traverse = rv_list2;
 
-	if (!rv_list1 || !rv_list2) {
-		LOG(LOG_ERROR, "Null arguments!\n");
+	if (!rv_list1_traverse || !rv_list2_traverse) {
+		LOG(LOG_ERROR, "Received NULL arguments\n");
 		goto end;
 	}
 
-	// FIXME: rv_lists are very unlikely to change, being static scenario,
-	// therefore detailed check on rv_list is skipped.
+	if (rv_list1_traverse->num_rv_directives != rv_list2_traverse->num_rv_directives) {
+		LOG(LOG_ERROR, "Number of RendezvousDirective(s) do not match\n");
+		goto end;
+	}
+	while (rv_directive_index < rv_list1_traverse->num_rv_directives &&
+		rv_directive_index < rv_list2_traverse->num_rv_directives) {
+		fdo_rendezvous_directive_t *directive1 =
+			fdo_rendezvous_directive_get(rv_list1_traverse, rv_directive_index);
+		fdo_rendezvous_directive_t *directive2 =
+			fdo_rendezvous_directive_get(rv_list2_traverse, rv_directive_index);
+		if (!directive1 || !directive2) {
+			LOG(LOG_ERROR, "One of the RendezvousDirective(s) is empty\n");
+			goto end;
+		}
+		rv_instr_index = 0;
+		while (rv_instr_index < directive1->num_entries &&
+			rv_instr_index < directive2->num_entries) {
+			fdo_rendezvous_t *entry_ptr1 =
+				fdo_rendezvous_list_get(directive1, rv_instr_index);
+			fdo_rendezvous_t *entry_ptr2 =
+				fdo_rendezvous_list_get(directive2, rv_instr_index);
+			if ((!entry_ptr1 || !entry_ptr2)) {
+				LOG(LOG_ERROR, "One of the RendezvousInstr(s) is empty\n");
+				goto end;
+			}
+			if (!fdo_rendezvous_instr_compare(entry_ptr1, entry_ptr2)) {
+				LOG(LOG_ERROR, "One of the RendezvousInstr(s) is empty\n");
+				goto end;				
+			}
+			rv_instr_index++;
+		}
+		rv_directive_index++;
+	}
 	retval = true;
-
 end:
 	return retval;
 }
 
+/**
+ * Compare the given RendezvousInstr(s) represented by the two fdo_rendezvous_t, with one another.
+ * 
+ * @param entry1: poniter to input first fdo_rendezvous_t object
+ * @param entry2: poniter to input seconf fdo_rendezvous_t object
+ * @return
+ *        true if both RendexvousInstr(s) are same else false.
+ */
+bool fdo_rendezvous_instr_compare(fdo_rendezvous_t *entry1, fdo_rendezvous_t *entry2) {
+
+	int memcmp_diff;
+
+	if (!entry1 || !entry2) {
+		LOG(LOG_ERROR, "Received NULL arguments\n");
+		return false;
+	}
+
+	if ((entry1->dev_only != NULL && entry2->dev_only != NULL &&
+		(*entry1->dev_only ==  *entry2->dev_only)) ||
+		(entry1->dev_only == NULL && entry2->dev_only == NULL)) {
+		return true;
+	} else {
+		LOG(LOG_ERROR, "RendezvousInstr: RVDEVONLY compare failed\n");
+		return false;
+	}
+
+	if ((entry1->owner_only != NULL && entry2->owner_only != NULL &&
```

Comment:
Is this code ever reached? It appears that control is returned with previous statements.